crates/btcio/src/rpc/types.rs

```diff
@@ -35,6 +35,49 @@ pub enum TransactionCategory {
     Orphan,
 }
 
+/// Result of JSON-RPC method `getblockchaininfo`.
+///
+/// Method call: `getblockchaininfo`
+///
+/// > Returns an object containing various state info regarding blockchain processing.
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct GetBlockchainInfo {
+    /// Current network name as defined in BIP70 (main, test, signet, regtest).
+    pub chain: String,
+    /// The current number of blocks processed in the server.
+    pub blocks: u64,
+    /// The current number of headers we have validated.
+    pub headers: u64,
+    /// The hash of the currently best block.
+    #[serde(rename = "bestblockhash")]
+    pub best_block_hash: String,
+    /// The current difficulty.
+    pub difficulty: f64,
+    /// Median time for the current best block.
+    #[serde(rename = "mediantime")]
+    pub median_time: u64,
+    /// Estimate of verification progress (between 0 and 1).
+    #[serde(rename = "verificationprogress")]
+    pub verification_progress: f64,
+    /// Estimate of whether this node is in Initial Block Download (IBD) mode.
+    #[serde(rename = "initialblockdownload")]
+    pub initial_block_download: bool,
+    /// Total amount of work in active chain, in hexadecimal.
+    #[serde(rename = "chainwork")]
+    pub chain_work: String,
+    /// The estimated size of the block and undo files on disk.
+    pub size_on_disk: u64,
+    /// If the blocks are subject to pruning.
+    pub pruned: bool,
+    /// Lowest-height complete block stored (only present if pruning is enabled).
+    #[serde(rename = "pruneheight")]
+    pub prune_height: Option<u64>,
+    /// Whether automatic pruning is enabled (only present if pruning is enabled).
+    pub automatic_pruning: Option<bool>,
+    /// The target size used by pruning (only present if automatic pruning is enabled).
+    pub prune_target_size: Option<u64>,
+}
+
```

Comment:
Note this was taken from `bitcoind_json_rpc_types::v26::GetBlockchainInfo` and adapted for v25-v28.